tests/staticfiles_tests/test_checks.py

```diff
@@ -132,3 +132,53 @@ def test_nonexistent_directories(self):
             # Nonexistent directories are skipped.
             finder = get_finder("django.contrib.staticfiles.finders.FileSystemFinder")
             self.assertEqual(list(finder.list(None)), [])
+
+
+class StorageCheckTests(SimpleTestCase):
+    def test_no_errors_without_setting(self):
+        errors = check_storages(None)
+        self.assertEqual(errors, [])
+
+    @override_settings(
+        STORAGES={
+            "default": {
+                "BACKEND": "django.core.files.storage.FileSystemStorage",
+            },
+            "staticfiles": {
+                "BACKEND": "django.contrib.staticfiles.storage.StaticFilesStorage",
+            },
+        }
+    )
+    def test_no_errors_staticfiles_and_default(self):
+        errors = check_storages(None)
+        self.assertEqual(errors, [])
+
+    @override_settings(
+        STORAGES={
+            "staticfiles": {
+                "BACKEND": "django.contrib.staticfiles.storage.StaticFilesStorage",
+            },
+        }
+    )
+    def test_no_errors_staticfiles_only(self):
+        errors = check_storages(None)
+        self.assertEqual(errors, [])
+
+    @override_settings(
+        STORAGES={
+            "default": {
+                "BACKEND": "django.core.files.storage.FileSystemStorage",
+            },
+        }
+    )
+    def test_error_missing_staticfiles(self):
+        errors = check_storages(None)
+        self.assertEqual(
+            errors,
+            [
+                Error(
+                    "You must define a 'staticfiles' storage in your STORAGES setting.",
+                    id="staticfiles.E005",
+                ),
+            ],
+        )
```

Comment:
This test fails, and I don't understand why. If I print the value of the `STORAGES` setting, it has both `default` and `staticfiles` in it. Am I doing it wrong?

Reply:
It turns out that `override_settings` has some backwards compatibility code which tries to preserve the other key in the setting:

https://github.com/django/django/blob/32940d390a00a30a6409282d314d617667892841/django/conf/__init__.py#L332-L341